fairmq/zeromq/FairMQMessageZMQ.h

```diff
@@ -31,26 +32,33 @@ class FairMQMessageZMQ : public FairMQMessage
     FairMQMessageZMQ(void* data, const size_t size, fairmq_free_fn* ffn, void* hint = nullptr);
     FairMQMessageZMQ(FairMQUnmanagedRegionPtr& region, void* data, const size_t size);
 
-    virtual void Rebuild();
-    virtual void Rebuild(const size_t size);
-    virtual void Rebuild(void* data, const size_t size, fairmq_free_fn* ffn, void* hint = nullptr);
+    void Rebuild() override;
+    void Rebuild(const size_t size) override;
+    void Rebuild(void* data, const size_t size, fairmq_free_fn* ffn, void* hint = nullptr) override;
 
-    virtual void* GetMessage();
-    virtual void* GetData();
-    virtual size_t GetSize();
+    void* GetMessage() override;
+    void* GetData() override;
+    size_t GetSize() const override;
 
-    virtual void SetMessage(void* data, const size_t size);
+    bool SetUsedSize(const size_t size) override;
+    void ApplyUsedSize();
 
-    virtual FairMQ::Transport GetType() const;
+    void SetMessage(void* data, const size_t size) override;
 
-    virtual void Copy(const std::unique_ptr<FairMQMessage>& msg);
+
+    FairMQ::Transport GetType() const override;
+
+    void Copy(const std::unique_ptr<FairMQMessage>& msg) override;
```

Comment:
At some point we should fix these `const unique_ptr &` signatures. `const FairMQMessage&` will do just fine, or? https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/

Reply:
agree, we should move away from that passing. Since this is breaking I will keep the old method and mark it as deprecated.

Reply:
Done. I updated only the Copy method. For Send/Receive I want to try what we discussed with passing with ownership transfer, but that would be a different PR.